File: Python/BubbleSort/BubbleSort.py
def BubbleSort(A):

    for i in range(len(A)):
        print('------')
        for j in range(len(A)-i-1):
            print(A[j], end=" ")
            print(A[j+1])
            print(A)
            if A[j]>A[j+1]:
                A[j],A[j+1]=A[j+1],A[j]
            print(A)

A = [6,4,2,9,1]

BubbleSort(A)

print(A)

File: Kotlin/src/BubbleSort.kt
fun main(){
    println("*********************** Bubble Sort *************************")
    val arr = arrayOf(9,8,1,2,6,5,4,3,2,7,0)
    println("Array: ${arr.contentToString()}, Sorted Array: ${bubbleSort(arr).contentToString()}")
}

fun bubbleSort(arr: Array<Int>): Array<Int>{
    for(i in arr.indices){ // 0 to arr.size-1
        for(j in 0..arr.size-2-i){
            if(arr[j] > arr[j+1]){
                arr[j] = arr[j+1].also { arr[j+1] = arr[j] }
            }
        }
    }
    return arr
}
